modules/fgbio/fastqtobam/main.nf

```diff
@@ -9,29 +9,30 @@ process FGBIO_FASTQTOBAM {
 
     input:
     tuple val(meta), path(reads)
-    val read_structure
 
     output:
-    tuple val(meta), path("*_umi_converted.bam"), emit: umibam
-    path "versions.yml"                         , emit: versions
+    tuple val(meta), path("*.bam"), emit: bam
+    path "versions.yml"           , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
+    def sample_name = args.contains("--sample") ? "" : "--sample ${prefix}"
+    def library_name = args.contains("--library") ? "" : "--library ${prefix}"
+
     """
 
     fgbio \\
         --tmp-dir=. \\
         FastqToBam \\
-        -i $reads \\
-        -o "${prefix}_umi_converted.bam" \\
-        --read-structures $read_structure \\
-        --sample $meta.id \\
-        --library $meta.id \\
-        $args
+        ${args} \\
```

Comment:
Would it make sense to include async I/O?

> --async-io[[=true|false]]     Use asynchronous I/O where possible, e.g. for SAM and BAM files. [Default: false]

Reply:
possibly, but I'd leave that option to `ext.args` since it might not be suitable for all filesystems